cupy/_environment.py

```diff
@@ -369,6 +375,23 @@ def _preload_library(lib):
                 _log(f'Library {lib} could not be preloaded: {e}')
 
 
+def _get_cutensor_from_wheel(
+        version: str, cuda: str) -> List[str]:
+    """
+    Returns the list of shared library path candidates.
+    """
+    import pkg_resources  # defer import # NOQA
+    try:
+        dist = pkg_resources.get_distribution(f'cutensor=={version}.*')
+    except pkg_resources.ResolutionError as e:
+        _log(f'cuTENSOR wheel could not be loaded: {type(e).__name__}: {e}')
+        return []
+    return [os.path.join(
+                dist.module_path, 'cutensor', 'lib', cuda_ver_path,
```

Comment:
Q: In the case that we split the `cutensor` wheel into per-CUDA version, would you like us to keep this directory structure?

Reply:
(Another note on the directory structure is it seems most of CUDA wheels are installed to the `nvidia` folder, but `cutensor` has not followed because no one asked.)

Reply:
I think it would be convenient if Python APIs to discover bundled files are added to cutensor package (`cutensor.get_include_dir()`, `cutensor.get_lib_dir()`, `cutensor.get_lib_files()` etc.)
For example, [certifi](https://pypi.org/project/certifi/), which provides a root CA certificate for TLS, contains a PEM file along with the `certifi.where()` API that returns the path to the file.  

`cutensor.__version__` will be also helpful. If this is implemented, we can check cuTENSOR version on our own, without using `pkg_resources` or `importlib.metadata`.

Reply:
> Q: In the case that we split the cutensor wheel into per-CUDA version, would you like us to keep this directory structure?

I don't have a specific requirement (we can just update the code 😄), but  `lib/libcutensor.so.1` sounds more natural to me than `lib/11/libcutensor.so.1`.

Reply:
Thanks for feedback. I will discuss with both cuQuantum and cuTENSOR teams and investigate alternative possibilities. 

We implemented exactly what you asked for in cuQuantum (internal repo), but it lives in the `cuquantum-python` wheel, not in the `cuquantum` wheel. The reason is that there is ambiguity in the behavior of your suggested APIs, so we had to disambiguate by letting Python do the import (during which the dynamic linker would load `libcutensor.so`, `libcutensornet.so`, etc), and inspect the linker's internal state. This ensures regardless of how cuTENSOR etc are installed (wheel, conda, tarball, etc), users calling these APIs will always get the correct path.

Obviously this only works on Linux, but for cuQuantum Python it is sufficient, and we can cover cuTENSOR as well. The other drawback is obviously this capability must stay in the Python binding, not with the binary-repackaged wheel. Thoughts are more than welcome...

Reply:
As a temporary solution until we land on something for the cuTENSOR wheel, I am thinking here you can just inspect `site.getsitepackages()` and `site.getusersitepackages()`. If cuTENSOR wheel is installed, it must live in one of the site-packages.

Reply:
Update:
> We implemented exactly what you asked for in cuQuantum (internal repo), but it lives in the `cuquantum-python` wheel, not in the `cuquantum` wheel. 

See https://github.com/NVIDIA/cuQuantum/blob/main/python/cuquantum/__main__.py.

> The reason is that there is ambiguity in the behavior of your suggested APIs, so we had to disambiguate by letting Python do the import (during which the dynamic linker would load `libcutensor.so`, `libcutensornet.so`, etc), and inspect the linker's internal state. 

Correction: There's no ambiguity in your suggested APIs. It's just that the wheel would no longer be simple binary repackaging, but instead it needs to come with a Python script like the above one living in the same directory. Since it incurs additional resources that do not yet exist in our packaging pipeline, it'd not be addressed in the near term 😅

The ambiguity I had in mind referred to the fact that there could be multiple `libcutensor.so` living in different places loaded in the same Python process. This is possible. It depends on how different modules were built and linked. Python allows this because Python by default loads a shared library in private mode (`RTLD_LOCAL`). It's fine for the purpose of a runtime, but not OK if we want to return the right path to a build system or user. Either your suggestion or our (cuQuantum) approach is needed.